### PR TITLE
Issues #252, #268, #269, #270, #271

### PR DESCRIPTION
**Pull request by vqmarkman**
Issue #268: new method of confirmation that page was loaded.
Issue #252: fix probes and labels tests where element can't be found in case of multiple label groups

### CODE REVIEW

**Comment by vqmarkman** [comment]
@susannahmarcus, I've implemented all loading solution for all of the reports.

**Remaining issues:**
- issue 288: queries report is randomly failing in unclear place
- issue 40: Clicking left sidebar for labels will return user to last visited page #40  (or I think it is the same issue)

My suggestion would be: let's fix everything we know about in this PR, make it stable and merge.
It seems to me that we are very close.

File: cypress/e2e/warehouse.spec.cy.js [hunk 1]
```diff
@@ -29,32 +30,27 @@ describe("Warehouse section", () => {
...
-    cy.get('li[role="option"]')
-        .should("be.visible")
+    clickCheck({ clickElem: 'li[role="option"]', contains: "Warehouse Activity" });
+
+    // Check that text "Warehouse Activity" is visible
+    cy.get("span")
         .contains("Warehouse Activity")
-        .should("be.visible")
-        .click();
+        .should("be.visible");
```
Comment (susannahmarcus) [inline cypress/e2e/warehouse.spec.cy.js:38]: there seems to be a click that was taken out here

File: cypress/support/utils.js [hunk 2]
```diff
@@ -131,13 +128,16 @@ export const fillInNewGroupedLabelForm = (
...
 export const addNewLabelToGroup = (groupName, labelName, condition, rank) => {
+
   // Find tab with the group name and click on it
   cy.get('button[data-baseweb="tab"')
     .should("exist")
     .find('div[data-testid="stMarkdownContainer"]')
     .should("exist")
     .contains(groupName)
-    .click();
+    .as("labelGroupTab");
```
Comment (susannahmarcus) [inline cypress/support/utils.js:138]: we might want to refactor this to use the as() earlier. sometimes, by the time the test reaches contains() there has been a rerender, so it's better to make an alias earlier on. I haven't looked at these tests yet, but if this point is flaky then that could be one thing we could try.

File: cypress/support/utils.js [hunk 3]
```diff
@@ -170,6 +171,17 @@ export const addNewLabelToGroup = (groupName, labelName, condition, rank) => {
 
 // For ungrouped label, specify "Ungrouped" in the groupName argument
 export const labelDelete = (groupName, labelName) => {
+  cy.log("*** labelDelete (begin): groupName:labelName", groupName,labelName);
```
Comment (susannahmarcus) [inline cypress/support/utils.js:174]: nit: do we want to keep this log?

File: cypress/support/utils.js [hunk 4]
```diff
@@ -181,8 +193,10 @@ export const labelDelete = (groupName, labelName) => {
       cy.get('div[data-testid="column"]')
         .contains("🗑️")
         .should("exist")
-        .click();
+        .click({ force: true });
     });
+
+  cy.log("*** labelDelete (end): groupName", labelName);
```
Comment (susannahmarcus) [inline cypress/support/utils.js:199]: nit: do we want to keep this log?

File: cypress/support/utils.js [hunk 5]
```diff
@@ -294,9 +308,7 @@ export const clickCheck = (options) => {
 
 export const checkForLoading = () => {
   cy.get('[data-testid="stMarkdownContainer"]')
-    .contains("Please wait...")
-    .should("not.exist", { timeout: 120000 });
-  cy.get('[data-testid="stStatusWidget"]').should("not.exist", {
-    timeout: 120000,
-  });
+    .contains("Please wait...", { timeout: 360000 })
```
Comment (susannahmarcus) [inline cypress/support/utils.js:311]: i _think_ we might want to move the timeout to the "get" not the contains. I'll check it locally once merged

**Review by susannahmarcus — APPROVED** [review]
there aqre some things we might wanna fix, but I'll look into them locally post merge, when I'm rebasing onto these changes

File: cypress/e2e/warehouse.spec.cy.js (hunk 1 above)
Comment (vqmarkman) [reply cypress/e2e/warehouse.spec.cy.js:38]: yea, the diff shows it in a weird way. There was a click that I replaced with a call to clickCheck() and then I follow up with checking if we can actually see "Warehouse Activity" (that might be redundant actually, but not harmful ... I think)

File: cypress/support/utils.js (hunk 2 above)
Comment (vqmarkman) [reply cypress/support/utils.js:138]: yes, you are absolutely correct! I've already refactored couple of calls like this for probes I believe.

File: cypress/support/utils.js (hunk 3 above)
Comment (vqmarkman) [reply cypress/support/utils.js:174]: will keep it for now while we are stabilizing things (easier to read cypress output)

File: cypress/support/utils.js (hunk 4 above)
Comment (vqmarkman) [reply cypress/support/utils.js:199]: will remove after we are done debugging.

File: cypress/support/utils.js (hunk 5 above)
Comment (vqmarkman) [reply cypress/support/utils.js:311]: not sure ... I felt that it needed to be waiting for "Please wait" to appear ... but we will see after the merge.
Thanks for the thorough review, @susannahmarcus ! Really appreciate it.

**Pull request merged**